### PR TITLE
 feat: expose a core bundle w/o parser

### PR DESCRIPTION
**Pull request by longlho**
If you're guaranteed to always provide `react-intl`, you can use our core bundle which is significantly smaller in size by swapping out `react-intl` `import`:

```tsx
// Before
import {IntlProvider} from 'react-intl';
// After
import {IntlProvider} from 'react-intl/lib/core'; // ES6 import
// OR
import {IntlProvider} from 'react-intl/dist/core'; // ES5 commonjs
```

Size difference:

| Package           | Minified Size | Minzipped Size |
| ----------------- | ------------- | -------------- |
| `react-intl`      | `29K`         | `9.07K`        |
| `react-intl` core | `19K`         | `6.32K`        |

### CODE REVIEW

**Comment by redonkulus** [comment]
Is there a better way to expose this without having to drill into the package directory path?

File: docs/API.md [hunk 1]
```diff
@@ -483,3 +460,46 @@ The React components provided by React Intl allow for a declarative, idiomatic-R
...
+};
+
+ReactDOM.render(
+  <IntlProvider messages={messages}>
+    <FormattedMessage id="ast_simple" />
+  </IntlProvider>
+); // will render `hello world`
+```
+
+If you're guaranteed to always provide `react-intl`, you can use our core bundle which is significantly smaller in size by swapping out `react-intl` `import`:
```
Comment (redonkulus) [inline docs/API.md]: What does this mean, guaranteed to provide react-intl? In what scenarios wouldn’t you?

File: src/components/message.tsx [hunk 2]
```diff
@@ -7,8 +7,7 @@
 import * as React from 'react';
 import withIntl from './injectIntl';
 import {MessageDescriptor, IntlShape} from '../types';
-import * as shallowEquals_ from 'shallow-equal/objects';
-const shallowEquals = shallowEquals_;
+const shallowEquals = require('shallow-equal/objects');
```
Comment (redonkulus) [inline src/components/message.tsx]: Why require?

Comment (longlho) [reply src/components/message.tsx]: bc those deps don't have typings & rollup complains if you `import * as foo` and then use `foo()` (namespace isn't technically a function)

**Pull request merged**